content/docs/examples/tutorial/setup-kubernetes-cluster/index.md

```diff
@@ -0,0 +1,208 @@
+        http:
+          paths:
+          - path: /
+            backend:
+              serviceName: prometheus
+              servicePort: 9090
+      - host: my-kiali.io
+        http:
+          paths:
+          - path: /
+            backend:
+              serviceName: kiali
+              servicePort: 20001
+    EOF
+    {{< /text >}}
+
+1.  You may want to limit the permissions of each participant so they will be able to create resources only in their
+    namespace. Even if you run the tutorial alone on your own cluster, it could be a good practice to limit yourself so
+    your learning will not interfere with other namespaces in your cluster.
+
+    Perform the following steps to achieve this:
+
+    1.  Create a role to provide read access to `istio-system` namespace:
+
+        {{< text bash >}}
+        $ kubectl apply -f - <<EOF
+        kind: Role
+        apiVersion: rbac.authorization.k8s.io/v1beta1
+        metadata:
+          name: istio-system-access
+          namespace: istio-system
+        rules:
+        - apiGroups: ["", "extensions", "apps"]
+          resources: ["*"]
+          verbs: ["get", "list"]
+        EOF
+        {{< /text >}}
+
+    1.  Create a service account for the user of the namespace and provide access to the namespace's resources to that
+    service account:
+
+        {{< text bash >}}
+        $ kubectl apply -f - <<EOF
+        apiVersion: v1
+        kind: ServiceAccount
+        metadata:
+          name: ${NAMESPACE}-user
+          namespace: $NAMESPACE
+        ---
+        kind: Role
+        apiVersion: rbac.authorization.k8s.io/v1beta1
+        metadata:
+          name: ${NAMESPACE}-access
+          namespace: $NAMESPACE
+        rules:
+        - apiGroups: ["", "extensions", "apps", "networking.k8s.io", "networking.istio.io", "authentication.istio.io",
+                      "rbac.istio.io", "config.istio.io"]
+          resources: ["*"]
+          verbs: ["*"]
+        ---
+        kind: RoleBinding
+        apiVersion: rbac.authorization.k8s.io/v1beta1
+        metadata:
+          name: ${NAMESPACE}-access
+          namespace: $NAMESPACE
+        subjects:
+        - kind: ServiceAccount
+          name: ${NAMESPACE}-user
+          namespace: $NAMESPACE
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: Role
+          name: ${NAMESPACE}-access
+        ---
+        kind: RoleBinding
+        apiVersion: rbac.authorization.k8s.io/v1beta1
+        metadata:
+          name: ${NAMESPACE}-istio-system-access
+          namespace: istio-system
+        subjects:
+        - kind: ServiceAccount
+          name: ${NAMESPACE}-user
+          namespace: $NAMESPACE
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: Role
+          name: istio-system-access
+        EOF
+        {{< /text >}}
+
+    1.  Generate the `./${NAMESPACE}-user-config.yaml` Kubernetes configuration file needed for each user of each
```

Comment:
The generated YAML is used by the KUBECONFIG environment variable to setup the context for kubectl command. Each user needs a different YAML since the YAML sets the user's credentials, permissions and the default namespace. So one user will have the default namespace as `tutorial1` and will have permissions only for `tutorial1`. Another user will have the default namespace as `tutorial2` and will have permissions only for `tutorial2`. Different YAML files will contain different default namespaces and permissions for them.

Reply:
The user does not apply the configuration in their namespace. The user apply the configuration and as a result their namespace is set for them as the default namespace.

Reply:
@rcaballeromx I rewrote a bit the steps in https://github.com/istio/istio.io/pull/4704/commits/f32215af80b14c15b2ac896f14baa06392546f22, added some clarifications, please take a look.